File: LeetCode/cpp/905.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int> &A) {
        int i = 0, j = 0;
        while (j < A.size()) {
            if (A[j] % 2 == 0) {
                swap(A[i], A[j]);
                ++i;
            }
            ++j;
        }
        return A;
    }
};

File: LeetCode/python/905.py
class Solution(object):
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        j = 0
        for i in range(len(A)):
            if(A[i] % 2 == 0):
                A[i], A[j] = A[j], A[i]
                j += 1
        return A